src/compiler/types.ts

```diff
@@ -3005,7 +3005,24 @@ namespace ts {
         isRestParameter?: boolean;
     }
 
-    export type SymbolTable = Map<Symbol>;
+    export type EscapedIdentifier = (string & { __escapedIdentifier: void }) | (void & { __escapedIdentifier: void });

```

Comment:
Also, re: the shape:

> The shape of this brand is also rather unique compared to others we've used. Instead of just an intersection of a string and an object, it is that union-ed with an intersection of void and an object. This makes it wholly incompatible with a normal string (which is good, it cannot be missued on assignment or on usage), while still being comparable with a normal string via === (also good) and castable from a string.